src/components/sparks/Sparks.spec.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {render} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {testA11y} from '../../axe';
+import Sparks from './Sparks';
+
+// mock all animations WAAPI so tests run correctly in jsdom environment (no window)
+jest.mock('./animation', () => {
+  return {
+    useAnimation: jest.fn(() => ({
+      register: jest.fn(),
+      phase: 'initial',
+      setPhase: jest.fn(),
+    })),
+  };
+});
+
+describe('<Sparks />', () => {
+  it('renders correctly and displays its children', () => {
+    const sparks = render(
+      <Sparks>
+        <button type="button">Click me</button>
+      </Sparks>
+    );
+
+    expect(sparks.container.firstChild).toBeTruthy();
+    expect(sparks.getByRole('button', {name: /click me/i})).toBeTruthy();
+  });
+
+  it('renders around interactive element and this element still captures all the events', () => {
```

Comment:
I would rephrase it a little bit 'renders around interactive element and this element still can be clicked' because we actually testing it but not all events.